### PR TITLE
Allow stake-address-info query command to accept reward account addresses

### LINKED ISSUE #1197: [FR] - JSON output for query stake-address-info (--output-format json)
Tag 1.13.0, Pioneer

Please provide the ability for cardano-cli to output all requested queries in a standard json format like using the --output-format json parameter with jcli.

Otherwise, please change the output of the stake-address-query request
`./cardano-cli shelley query stake-address-info `
into a standard json format output.

But a better solution would be to force the cli to output json with an additional parameter.

Or if there is a pipeable parser available that parses the current outputs into json format, please let us know.

Thx!

### PR DESCRIPTION
**Pull request by intricate**
This also closes #1197.

### CODE REVIEW

**Comment by dcoutts** [comment]
I think in fact we only need the command to work for stake addresses. It is not necessary (and probably confusing) to have it work for payment addresses that are associated with a stake address.

**Comment by intricate** [comment]
> I think in fact we only need the command to work for stake addresses. It is not necessary (and probably confusing) to have it work for payment addresses that are associated with a stake address.

Oops. That's my mistake in the first place.

**Comment by gitmachtl** [comment]
Hi, can we please have a json output format for the "stake-address-info query"? Or an even better solution would be to have the --output-format json option for all the cli outputs. Thx!

P.S.: If there is a parser already available that do a conversion from your format into json, please let me know.

File: cardano-api/src/Cardano/Api/Convert.hs [hunk 1]
```diff
@@ -32,9 +32,28 @@ addressFromHex :: Text -> Either Binary.DecoderError Address
 addressFromHex txt =
   case Base16.decode (Text.encodeUtf8 txt) of
     (raw, _) ->
-      case Shelley.deserialiseAddr raw of
-        Just addr -> Right $ AddressShelley addr
-        Nothing ->  AddressByron <$> Binary.decodeFull' raw
+      tryDeserialise
+        raw
+        [ maybe (Left addrShelleyErr) (Right . AddressShelley) . Shelley.deserialiseAddr
```
Comment (Jimbo4350) [inline cardano-api/src/Cardano/Api/Convert.hs:37]: Would this be more readable as three separate functions in the `where` clause?

File: cardano-api/src/Cardano/Api/LocalStateQuery.hs [hunk 2]
```diff
@@ -195,27 +229,29 @@ queryLocalLedgerState network socketPath point = do
...
-  let pointAndQuery = (point, GetFilteredDelegationsAndRewardAccounts creds)
-  newExceptT $ liftIO $
-    queryNodeLocalState
-      nullTracer
-      (pClientInfoCodecConfig . protocolClientInfo $ mkNodeClientProtocolTPraos)
-      network
-      socketPath
-      pointAndQuery
+  -> ExceptT LocalStateQueryError IO DelegationsAndRewards
+queryDelegationsAndRewardsFromLocalState network socketPath addrs point = do
```
Comment (Jimbo4350) [inline cardano-api/src/Cardano/Api/LocalStateQuery.hs:238]: :+1:

**Comment by dcoutts** [comment]
bors merge

**Pull request merged**